### PR TITLE
refactor: extract VSMatches function

### PR DESCRIPTION
**Pull request by wulianglongrd**
Minor refactoring:
1. Extract the VS matching logic into `VSMatches` function to be consistent with `Service` matching.

https://github.com/istio/istio/blob/e011f1aec5cdcc0780768aaa31328cad1412f6e8/pilot/pkg/model/sidecar.go#L66

2. Determine whether to use gateway semantics before the for loop (see L46). No need to determine in every loop.

### CODE REVIEW

File: pilot/pkg/model/sidecar.go [hunk 1]
```diff
@@ -85,6 +85,27 @@ func (hc hostClassification) Matches(h host.Name) bool {
 	return false
 }
 
+func (hc hostClassification) VSMatches(vsHost host.Name, useGatewaySemantics bool) bool {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go:88]: please add comments

**Review by hzxuzhonghu — COMMENTED** [review]
we also have `vsHostMatches`, which makes vs matching confusing, i would suggest we keep one function. maybe inline

File: pilot/pkg/model/sidecar.go (hunk 1 above)
Comment (wulianglongrd) [reply pilot/pkg/model/sidecar.go:90]: Thanks, added.

**Review by hzxuzhonghu — APPROVED** [review]
tks

**Pull request merged**